support-frontend/assets/pages/contributions-landing/components/StripePaymentRequestButton/StripePaymentRequestButton.jsx

```diff
@@ -263,23 +246,28 @@ function onPayment(
   // We need to do this so that we can offer marketing permissions on the thank you page
   updatePayerEmail(paymentRequestData, props.updateEmail);
 
-  const isUsOrCanadian = props.countryGroupId === UnitedStates || props.countryGroupId === Canada;
-
-  const stateOrProvinceUpdateOk = isUsOrCanadian ?
-    updatePayerStateOrProvince(stateOrProvinceFromCard, props.stateOrProvince, props.updateStateOrProvince) : true;
-
-  // We need to update the country so it matches the state taken from the card
-  if (isUsOrCanadian && billingCountryFromCard) {
-    const isoCountry = countryFromString(billingCountryFromCard);
-    if (isoCountry) {
-      props.updateBillingCountry((isoCountry));
+  const validatedCountryFromCard: Option<IsoCountry> = findIsoCountry(billingCountryFromCard);
+
+  // If recurring, Zuora and Salesforce need a valid state or province code for US and Canada billing countries.
+  let countryAndStateUpdateOk =
+    props.contributionType !== 'ONE_OFF' ? !['US', 'CA'].includes(validatedCountryFromCard) : true;
+  if (validatedCountryFromCard) {
+    props.updateBillingCountry(validatedCountryFromCard);
+    const validatedBillingStateOrProvinceFromCard: Option<StateProvince> =
+      stateProvinceFromString(validatedCountryFromCard, billingStateOrProvinceFromCard);
+    if (validatedBillingStateOrProvinceFromCard) {
+      props.updateStateOrProvince(validatedBillingStateOrProvinceFromCard);
+      countryAndStateUpdateOk = true;
+    } else if (!props.stateOrProvince) {
+      logException('Missing address_state in payment request token and no state/province selected in the form');
     }
   }
 
-  const nameUpdateOk: boolean = props.stripeAccount !== 'ONE_OFF' ?
+  // If recurring, Salesforce needs a valid last name.
+  const nameUpdateOk: boolean = props.contributionType !== 'ONE_OFF' ?
     updatePayerName(paymentRequestData, props.updateFirstName, props.updateLastName) : true;
```

Comment:
```suggestion
  const nameUpdateOk: boolean = props.contributionType === 'ONE_OFF' || updatePayerName(paymentRequestData, props.updateFirstName, props.updateLastName);
```

Reply:
I did think of doing it that way, however it then inverts the logic to the comment above. If you feel it's better I'll do it this way round for both the `...Ok` variables?

Reply:
I think it works better with the name of the variable and is simpler, maybe tweak the comment slightly if you think it's misleading?